cluster-autoscaler/utils/scheduler/scheduler.go

```diff
@@ -66,3 +66,12 @@ func CreateNodeNameToInfoMap(pods []*apiv1.Pod, nodes []*apiv1.Node) map[string]
 
 	return nodeNameToNodeInfo
 }
+
+// NodeWithPod function returns NodeInfo, which is a copy of nodeInfo argument with an additional pod scheduled on it.
+func NodeWithPod(nodeInfo *schedulercache.NodeInfo, pod *apiv1.Pod) *schedulercache.NodeInfo {
+	podsOnNode := nodeInfo.Pods()
+	podsOnNode = append(podsOnNode, pod)
+	newNodeInfo := schedulercache.NewNodeInfo(podsOnNode...)
+	newNodeInfo.SetNode(nodeInfo.Node())
```

Comment:
I'm unsure about the error handling in this line. Do you think that I should handle the error and propagate in this func?

Reply:
Maybe just check if it returned error. And add logging here. Currently it does not return error ever. It is possible that the fact that `SetNode` returns an error object at all is a design mistake.

Reply:
My concern was that if SetNode() changes to return an error, we would miss it here. I'll add logging now.

Reply:
I think you missed the log here.